source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRImageRenderer.mm

```diff
@@ -39,9 +39,54 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
     std::shared_ptr<BaseCardElement> elem = [acoElem element];
     std::shared_ptr<Image> imgElem = std::dynamic_pointer_cast<Image>(elem);
     ACRUIImageView *view;
+    NSInteger pixelWidth = imgElem->GetPixelWidth(), pixelHeight = imgElem->GetPixelHeight();
+    BOOL hasExplicitMeasurements = (pixelWidth || pixelHeight);
+    BOOL isAspectRatioNeeded = !(pixelWidth && pixelHeight);
     CGSize cgsize = [acoConfig getImageSize:imgElem->GetImageSize()];
+
+    NSMutableDictionary *imageViewMap = [rootView getImageMap];
+    // Syncronize access to imageViewMap
+    NSNumber *number = [NSNumber numberWithUnsignedLongLong:(unsigned long long)imgElem.get()];
+    NSString *key = [number stringValue];
+    UIImage *img = imageViewMap[key];
+
+    CGFloat heightToWidthRatio = 0.0f, widthToHeightRatio = 0.0f;
+    if(img){
+        if(img.size.width > 0) {
+            heightToWidthRatio = img.size.height / img.size.width;
+        }
+
+        if(img.size.height > 0) {
+            widthToHeightRatio =img.size.width / img.size.height;
+        }
+    }
+
+    if(hasExplicitMeasurements) {
+        if(pixelWidth){
+            cgsize.width = pixelWidth;
+            if(isAspectRatioNeeded) {
+                cgsize.height = pixelWidth * heightToWidthRatio;
+            }
+        }
+        if(pixelHeight){
+            cgsize.height = pixelHeight;
+            if(isAspectRatioNeeded) {
+                cgsize.width = pixelHeight / heightToWidthRatio;
```

Comment:
>heightToWidthRatio [](start = 45, length = 18)

`widthToHeightRatio`

Reply:
But if the image has zero width or height, this will throw.

---
In reply to: [196895809](https://github.com/Microsoft/AdaptiveCards/pull/1374#discussion_r196895809) [](ancestors = 196895809,196895492)

Reply:
thanks, i thought i had the check.

---
In reply to: [196896587](https://github.com/Microsoft/AdaptiveCards/pull/1374#discussion_r196896587) [](ancestors = 196896587,196895809,196895492)